### PR TITLE
[update-engine] add an optional message for success operations

### PR DESCRIPTION
**Pull request by sunshowers**
Also make metadata optional since in the majority of cases we aren't
returning any metadata.

With this, the new states and their requirements are:

| **kind** | **message** |     **metadata**     |
|:--------:|:-----------:|:--------------------:|
| success  | optional    | optional, completion |
| warning  | required    | optional, completion |
| skipped  | required    | optional, skipped    |

To express all of these separate cases, we have to introduce new
`StepSuccess`, `StepWarning` and `StepSkipped` builders.

Screenshot of a message attached to success:

image

### CODE REVIEW

**Review by jgallagher — APPROVED** [review]
LGTM :+1:

**Comment by sunshowers** [comment]
Ah looks like my cherry-pick didn't go well. It's based on #3194 so I'll just wait until that is landed before looking at this.

**Pull request merged**
